core/src/main/scala/akka/kafka/internal/ProducerStage.scala

```diff
@@ -43,7 +37,7 @@ class ProducerSendFlowStage[K, V](producerProvider: () => KafkaProducer[K, V])
       var completionState: Option[Try[Unit]] = None
 
       def checkForCompletion() = {
```

Comment:
you mean a timeout? I hope Kafka has a timeout for the `commitAsync` calls, which means that `awaitingConfirmation` will always be decremented down to 0.
Do you mean an addition timeout for the stage itself, yes we could have that.